packages/nuxt/src/pages/utils.ts

```diff
@@ -313,7 +316,7 @@ export async function getRouteMeta (contents: string, absolutePath: string, extr
     })
   }
 
-  metaCache[absolutePath] = extractedMeta
+  metaCache[cacheKey] = extractedMeta
   return extractedMeta
```

Comment:
would it resolve it just to do something like:

```suggestion
  return JSON.parse(JSON.stringify(extractedMeta))
```

Reply:
```
  if (absolutePath in metaCache && metaCache[absolutePath]) {
    return metaCache[absolutePath]
  }
```

I think it's the reason here? Has `key of absolutePath` cache, `return the first cache`?

Because the file is the same by `page:extends`?

Reply:
yes, we'd also have to update it there too

Reply:
Yes, you are right. Thank you very much. I will update it.

Reply:
Oh.. `extraxtedMeta` has `Set` properties. So, `JSON.stringify` has some problem..

Reply:
we could use klona

Reply:
I have changed this. Please check it out. Thank you very much. ❤

Reply:
you had some nice tests in the previous commit…. is it worth keeping them?